components/avatar/avatar.vue

```diff
@@ -150,6 +177,30 @@ export default {
       return element?.tagName?.toUpperCase() === 'IMG';
     },
 
+    randomizeGradientAngle () {
+      const angles = [45, -45, 90, -90, 135, -135, 180, -180];
+      return getRandomElement(angles);
+    },
+
+    randomizeGradientColorStops () {
+      const colors = new Set();
+      const colorsWith100 = ['purple-100', 'magenta-100', 'gold-100', 'blue-100'];
+      const colorsWith200 = ['purple-200', 'magenta-200', 'gold-200', 'blue-200'];
+
+      // get 3 unique colors, 2 from colorsWith100 and one from colorsWith200
+      while (colors.size < 3) {
+        if (colors.size === 2) {
+          colors.add(getRandomElement(colorsWith200));
+        } else {
+          colors.add(getRandomElement(colorsWith100));
+        }
+      }
+
+      const shuffledColors = Array.from(colors).sort(() => 0.5 - Math.random());
```

Comment:
@francisrupert can you validate this logic is okay? We obtain 2 color stops, two `100` and one `200` in no particular order, which means 100 and 200 color stops can be assigned to any `--avatar-gradient-stop-[1..3]` CSS custom property. That certainly will give us a broader spectrum of variants, right?

Reply:
Yesh, just perfect. Sweet shuffles. At first I thought it might be a challenge to prevent reusing a color (e.g. don't allow `purple-100` to be used twice), but it appears its simplicity inhibits it: "randomly choose 2 from colorsWith100".

I obviously defer to your judgement on poking holes in it, and of course @braddialpad.

I'm actually curious on the math of this, I'm probably doing this wrong... `(4^2)*((4^2)/2)=128`... doesn't feel right, there's gotta be more than 128 combinations.

Reply:
Nice use of `Set()` here to prevent duplicates. I think the logic here should work totally fine. This could possibly be made more dynamic but not sure it's necessary unless we are planning on changing number of colors and number of stops all the time

Reply:
I tried refactoring this to get rid of the sort and it just ended up being uglier than what is currently here.